_extension/src/client.ts

```diff
@@ -108,12 +109,17 @@ export class Client {
             },
         };
 
-        this.client = new LanguageClient(
-            "typescript.native-preview",
-            "typescript.native-preview-lsp",
-            serverOptions,
-            this.clientOptions,
-        );
+		this.client = new LanguageClient(
+			"typescript.native-preview",
+			"typescript.native-preview-lsp",
+			serverOptions,
+			{
+				...this.clientOptions,
+				initializationOptions: {
+					customConfigFileName,

```

Comment:
Rather than be part of initialization, we think this sort of thing would belong more on user preferences ([see here](https://github.com/microsoft/typescript-go/pull/1729).

Whether we just say "you need to restart the server" or we track this sort of thing on the server in some way is a bigger question.